packages/datetime/src/dateInput.tsx

```diff
@@ -50,6 +51,12 @@ export interface IDateInputProps extends IDatePickerBaseProps, IProps {
     closeOnSelection?: boolean;
 
     /**
+     * Props to pass to ReactDayPicker.
+     * See API documentation [here](http://react-day-picker.js.org).
```

Comment:
This seems like a better page to link to: http://react-day-picker.js.org/docs/api-daypicker.html

I'd also like to mention the props that we override and that won't be honored, to avoid confusion. The problem is that this is a fairly long list...

For `DateRangePicker`:

```
- canChangeMonth
- captionElement
- fromMonth
- month
- onMonthChange
- toMonth
```

For `DatePicker`:

```
- canChangeMonth
- captionElement
- disabledDays
- enableOutsideDays
- fromMonth
- locale
- localeUtils
- modifiers
- month
- onDayClick
- onMonthChange
- selectedDays
- toMonth
```

We can invoke the custom callbacks (`onDayClick`, `onMonthChange`) from our own `this.handle...` callbacks, so we wouldn't need to include those in the "blacklist" language.

We can also move some of the existing props _above_ the `dayPickerProps` spread to make our defaults overridable where it makes sense (I guess I'm mostly looking at `enableOutsideDays` per #1574). So that prop also needn't be mentioned. 

We would ideally mention the rest as being props you should either provide via the top-level `this.props` interface—or that won't be respected at all.

Finally, while this approach seems like the right way to support extensibility, I'm curious if it'll allow consumers to totally bork the component and get it out of sync with our Blueprint component (e.g. how would `props.dayPickerProps.initialMonth` interact with `props.initialMonth`?).

// CC @giladgray for thoughts.